crates/core/component/dex/src/component/router/path_search.rs

```diff
@@ -48,14 +54,14 @@ pub trait PathSearch: StateRead + Clone + 'static {
 
         let entry = cache.lock().0.remove(&dst);
         let Some(PathEntry { path, spill, .. }) = entry else {
+            record_duration();
             return Ok((None, None));
         };
```

Comment:
:point_up: this looks like an arm we should still record the duration of. even if we didn't find a path to return, we still searched for one.

i'd like to put a counter here and in some other arms below, in a subsequent branch.